src/Storage/Controllers/ProcessController.cs

```diff
@@ -80,28 +135,57 @@ public async Task<ActionResult<Instance>> PutProcess(
                 return NotFound();
             }
 
-            string taskId = null;
-            string altinnTaskType = existingInstance.Process?.CurrentTask?.AltinnTaskType;
+            var (action, taskId) = ActionMapping(processState, existingInstance);
 
-            if (processState?.CurrentTask?.FlowType == "AbandonCurrentMoveToNext")
+            bool authorized = await _authorizationService.AuthorizeInstanceAction(existingInstance, action, taskId);
+
+            if (!authorized)
             {
-                altinnTaskType = "reject";
+                return Forbid();
             }
-            else if (processState?.CurrentTask?.FlowType is not null
-                && processState.CurrentTask.FlowType != "CompleteCurrentMoveToNext")
+
+            UpdateInstance(existingInstance, processState, out var updateProperties);
+
+            Instance updatedInstance = await _instanceRepository.Update(existingInstance, updateProperties);
+
+            if (processState?.CurrentTask?.AltinnTaskType == "signing")
             {
-                altinnTaskType = processState.CurrentTask.AltinnTaskType;
-                taskId = processState.CurrentTask.ElementId;
+                await _instanceEventService.DispatchEvent(InstanceEventType.SentToSign, updatedInstance);
             }
 
-            string action = altinnTaskType switch
+            updatedInstance.SetPlatformSelfLinks(_storageBaseAndHost);
+            return Ok(updatedInstance);
+        }
+        
+        /// <summary>
+        /// Updates the process state of an instance.
+        /// </summary>
+        /// <param name="instanceOwnerPartyId">The party id of the instance owner.</param>
+        /// <param name="instanceGuid">The id of the instance that should have its process updated.</param>
+        /// <param name="processStateUpdate">The new process state of the instance (including instance events).</param>
+        /// <returns></returns>
+        [Authorize]
+        [HttpPut("batch")]
```

Comment:
Are we happy with the API name? Struggled to come up with alternatives here. It is a batch in the sense that there are multiple things (and multiple events). If the principle of modelling was behavior, and not resource/entity, it could be something like `CommitProcessTransition` or something like that, but that's not the case here..

Repo name is also less then ideal, but internal and easily changed.

cc @bjorntore @SandGrainOne

Reply:
I would prefer something other than "batch" if it was up to me.

When I see batch I'm thinking I can supply a list of multiple things that should be processed in bulk. This endpoint seems to me as a pretty normal PUT operation that changes some data related to an instance. My idea of a batch might not be very nuanced though.

Hard to come up with good suggestions. 🤔 

I think I'm leaning towards keeping same URL as old endpoint, just with v2: 
`storage/api/v2/instances/{instanceOwnerPartyId:int}/{instanceGuid:guid}/process`

Other suggestions? @SandGrainOne

Reply:
So, the options are:
* We include all current APIs in a v2, where this one endpoint changes. We risk making a lot of different API versions over time if we set this precedent
* We include only this changed API in a v2. What will become the new precedent? New APIs - will these be added to v1? v2? Both? Changed APIs - add only to  latest (v2)? 
* Find a decent name for the new endpoint and sticking with v1

I'm a fan of sticking to v1 here becuase building a strategy for API versioning is more work then I'd like to take on atm. It's probably something that would have to be discussed at the architecture forums as ideally all Altinn 3 teams have a cohesive strategy for public APIs (which includes how to do versioning)

Reply:
We can use a name similar to the naming of the new repository class: `instanceandevents`
```suggestion
        [HttpPut("instanceandevents")]
```